src/libmain/loggers.cc

```diff
@@ -21,7 +22,17 @@ LogFormat parseLogFormat(const std::string & logFormatStr) {
 }
 
 Logger * makeDefaultLogger() {
-    switch (defaultLogFormat) {
+    auto logFormat = defaultLogFormat;
+    std::string settingsLogFormat = settings.logFormat.get();
+    if (settingsLogFormat != "")
+    {
+        try {
+            logFormat = parseLogFormat(settingsLogFormat);
+        } catch (Error) {
```

Comment:
This should be `Error &`. But I don't think we should catch this error at all. We shouldn't silently ignore bad values.

Reply:
My thinking was that this would at least allow nix to run in a situation where the config is invalid, so user's systems don't get "stuck". I suppose there are easier ways of solving that though!

Reply:
CMD line option overrides config file options. So --log-format should be able to fix that.